stag-library/src/test/java/com/vimeo/stag/KnownTypeAdaptersTest.java

```diff
@@ -450,8 +452,36 @@ public void testForPrimitiveBooleanTypeAdapter() throws Exception {
 
         // call the TypeAdapter#read method
         boolean readValue = KnownTypeAdapters.PrimitiveBooleanTypeAdapter.read(new JsonReader(new StringReader(jsonString)), false);
-
         Assert.assertEquals(value, readValue);
+
+        String booleanString = "true";
+        // call the TypeAdapter#read method
+        readValue = KnownTypeAdapters.PrimitiveBooleanTypeAdapter.read(new JsonReader(new StringReader(booleanString)), false);
+        Assert.assertEquals(true, readValue);
+
+        booleanString = "false";
+        // call the TypeAdapter#read method
+        readValue = KnownTypeAdapters.PrimitiveBooleanTypeAdapter.read(new JsonReader(new StringReader(booleanString)), false);
+        Assert.assertEquals(false, readValue);
+
+        booleanString = "abcde"; // not valid, throw exception
+        // call the TypeAdapter#read method
+        readValue = KnownTypeAdapters.PrimitiveBooleanTypeAdapter.read(new JsonReader(new StringReader(booleanString)), false);
+        Assert.assertEquals(false, readValue);
+
+        booleanString = "1"; // 1 is treated as true
+        // call the TypeAdapter#read method
+        readValue = KnownTypeAdapters.PrimitiveBooleanTypeAdapter.read(new JsonReader(new StringReader(booleanString)), false);
+        Assert.assertEquals(true, readValue);
+
+        booleanString = "0"; // 0 is treated as false
+        // call the TypeAdapter#read method
+        readValue = KnownTypeAdapters.PrimitiveBooleanTypeAdapter.read(new JsonReader(new StringReader(booleanString)), false);
+        Assert.assertEquals(false, readValue);
```

Comment:
Use `Assert.assertFalse`, also we should pass in `true` as the default value for this read so that we know that the default value isn't being used